native.js

```diff
@@ -25,6 +25,13 @@ function asyncImpl(returnKind, promise) {
   $.pause(asyncImplStringAsync);
 }
 
+function preemptingImpl(returnKind, returnValue) {
```

Comment:
Note: I'm not currently using a counter as a cheaper preemption check because we don't have enough calls into the method to always preempt in time for some writes. In a future version we should investigate using a counter and not do the time check on every call (similar to how the disabled preemption code works in baseline.ts). Whether we inline the counter in each preemptingImpl call or put it in here is also debatable.